Meta/check-newlines-at-eof.py

```diff
@@ -0,0 +1,73 @@
+#!/bin/python3
+
+import os
+import subprocess
+import sys
+
+try:
+    os.chdir(os.path.dirname(__file__) + "/..")
+except:
+    os.exit(1)
+
+files = subprocess.run(
+    [
+        "git", "ls-files", "--",
+        "*.cpp",
+        "*.h",
+        "*.gml",
+        "*.html",
+        "*.js",
+        "*.css",
+        "*.sh",
+        "*.py",
+        ":!:Base",
+        ":!:Kernel/FileSystem/ext2_fs.h",
+        ":!:Libraries/LibC/getopt.cpp",
+        ":!:Libraries/LibCore/puff.h",
+        ":!:Libraries/LibCore/puff.cpp",
+        ":!:Libraries/LibELF/exec_elf.h"
+    ],
+    capture_output=True
+).stdout.decode().split('\n')
+
+MISSING_NEWLINE_AT_EOF_ERRORS = []
+MORE_THAN_ONE_NEWLINE_AT_EOF_ERRORS = []
+
+did_fail = False
+for filename in files:
+    try:
+        f = open(filename, "r")
+    except:
+        continue
+
+    f.seek(0, os.SEEK_END)
```

Comment:
I don't think a freshly opened file is ever not at seek position 0?

Reply:
This goes to the end of the file.

Reply:
...right. sorry. :facepalm: